Building a user interface for your contract

qbzzt opened the pull request:
Using modern components such as TypeScript, React, Vite, and Wagmi, we will go over a modern, but minimal, user interface and learn how to connect a wallet to the user interface, call a smart contract to read information, send a transaction to a smart contract, and monitor events from a smart contract to identify changes.

## Description

The completed article

## Related Issue

Closing #10489

nnnnnnsssss commented:
Recheck

qbzzt commented:
@wackerow , any progress on this? Sorry, but it's been open since November.

wackerow reviewed (approved):
@qbzzt Apologies for the delay to get this reviewed, but this looks great!

I added a couple commits to update it with `dev` and get the page rendering (few technical debt things we still need to clean up for new pages since the Next.js migration). 

The path name was `-for-you-contract` so I updated that to `-for-your-contract`. Also updated the heading levels... this pages are set up to only allow a single `<h1>` on any page, and for tutorials this is pulled from the `title` front matter property. As such, we can't use any single `#` in the markdown itself, so I bumped all of these up one level as the simplest fix.

Lastly, added `sidebarDepth: 3` to the front matter so the table of contents is more useful (shows deeper), especially after bumping the heading levels. 

Approving and gonna pull this in!

If we want, we could work on cleaning up the headers separately so they don't go to `h5`, but I don't see that as a reason to hold this up any longer in the meantime. Nice job! 🙏

qbzzt commented:
Thank you

On Wed, Mar 13, 2024 at 5:48 PM Paul Wackerow ***@***.***>
wrote:

> Merged #11682
> <https://github.com/ethereum/ethereum-org-website/pull/11682> into dev.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ethereum/ethereum-org-website/pull/11682#event-12110036551>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADBCGGOZWC4ET6NW2YCC34DYYDJVPAVCNFSM6AAAAAA7QOSZOCVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJSGEYTAMBTGY2TKMI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>

qbzzt commented:
> @qbzzt Apologies for the delay to get this reviewed, but this looks great!
> 
> I added a couple commits to update it with `dev` and get the page rendering (few technical debt things we still need to clean up for new pages since the Next.js migration).
> 
> The path name was `-for-you-contract` so I updated that to `-for-your-contract`. Also updated the heading levels... this pages are set up to only allow a single `<h1>` on any page, and for tutorials this is pulled from the `title` front matter property. As such, we can't use any single `#` in the markdown itself, so I bumped all of these up one level as the simplest fix.
> 
> Lastly, added `sidebarDepth: 3` to the front matter so the table of contents is more useful (shows deeper), especially after bumping the heading levels.
> 
> Approving and gonna pull this in!
> 
> If we want, we could work on cleaning up the headers separately so they don't go to `h5`, but I don't see that as a reason to hold this up any longer in the meantime. Nice job! 🙏

Thank you! I'll get back to the types of storage article (https://github.com/ethereum/ethereum-org-website/issues/6491).

qbzzt commented:
What about the coderabbitai suggestions? Should I go over them and see which ones make sense, and submit as a separate PR?

wackerow commented:
Not a requirement, but if you see any that you'd like to incorporate by all means!
